Feat/add ios version management (#1914)
saucelabs/sauce-docs: docs/mobile-apps/features/ios-version-management.md

---
id: ios-version-management
title: iOS Version Management
sidebar_label: iOS Version Management
description: Learn how to use iOS Version Management to test your app on the latest versions.
---

import useBaseUrl from '@docusaurus/useBaseUrl';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

:::info
This feature is only available for iOS Simulators.
:::

> With every major iOS release (e.g. iOS 15 to iOS 16) it takes 4-6 months for 80% of the end-users to fully adopt a new major version. To help you test your app on the latest versions of iOS, Sauce Labs provides iOS Version Management based on aliases. These aliases allow you to test your app on the latest versions of iOS without having to update your capabilities every time a new version is released.

Author: ```suggestion
With every major iOS release (for example iOS 15 to iOS 16) it takes 4-6 months for 80% of the end-users to fully adopt a new major version. To help you test your app on the latest versions of iOS, Sauce Labs provides iOS Version Management based on aliases. These aliases allow you to test your app on the latest versions of iOS without having to update your capabilities every time a new version is released.
```

capabilities.setCapability("sauce:options", sauceOptions);
```

</TabItem>
</Tabs>
</TabItem>
<TabItem value="js">

<Tabs
groupId="capability-js-emusim"
defaultValue="iphone"
values={[
{label: 'iPhone', value: 'iphone'},
{label: 'iPad', value: 'ipad'},
]}>
<TabItem value="iphone">

<!-- prettier-ignore -->
```js
const capabilities = {
    browserName: 'safari',
    platformName: 'ios',
    // Use one of the following aliases to test on the latest version of iOS of your choice
    'appium:platformVersion': 'latest|current_major|previous_major',
    'appium:deviceName': 'iPhone Simulator',
    'appium:automationName': 'xcuitest',
    'sauce:options': {
    }
}
```

</TabItem>
<TabItem value="ipad">

<!-- prettier-ignore -->
```js
const capabilities = {
    browserName: 'safari',
    platformName: 'ios',
    // Use one of the following aliases to test on the latest version of iOS of your choice
    'appium:platformVersion': 'latest|current_major|previous_major',
    'appium:deviceName': 'iPad Simulator',
    'appium:automationName': 'xcuitest',
    'sauce:options': {
    }
}
```

</TabItem>
</Tabs>

</TabItem>
<TabItem value="python">

<Tabs

> groupId="capability-python-emusim"

Reviewer: Does not render as proper md file, is this an error, or this will be used by SL help website UI to format it properly?

Author: This will be rendered properly by the [Docusaurus](https://docusaurus.io/)

capabilities.AddAdditionalCapability("appium:deviceName", "iPhone Simulator");
capabilities.AddAdditionalCapability("appium:automationName", "xcuitest");

HashMap<String, Object> sauceOptions = new Dictionary<string, object>();
capabilities.AddAdditionalCapability("sauce:options", sauceOptions);
```

</TabItem>
<TabItem value="ipad">

<!-- prettier-ignore -->
```csharp
AppiumOptions capabilities = new AppiumOptions();

capabilities.AddAdditionalCapability("browserName", "safari");
capabilities.AddAdditionalCapability("platformName", "ios");
// Use one of the following aliases to test on the latest version of iOS of your choice
capabilities.AddAdditionalCapability("appium:platformVersion", "latest|current_major|previous_major");
capabilities.AddAdditionalCapability("appium:deviceName", "iPad Simulator");
capabilities.AddAdditionalCapability("appium:automationName", "xcuitest");

HashMap<String, Object> sauceOptions = new Dictionary<string, object>();
capabilities.AddAdditionalCapability("sauce:options", sauceOptions);
```

</TabItem>
</Tabs>

</TabItem>
</Tabs>

## Examples

:::info

> The following examples use iOS 16 and 15 as the current and previous major versions of iOS, respectively. The examples don't always reflect the actual versions of iOS that are supported by Sauce Labs and Apple on simulators. For the most up-to-date information, see our [Platform Configurator](https://saucelabs.com/platform/platform-configurator#/) tool.

Author: ```suggestion
The following examples use iOS 16 and 15 as the current and previous major versions of iOS, respectively. The examples don't always reflect the actual versions of iOS that are supported by Sauce Labs and Apple on Simulators. See our [Platform Configurator](https://saucelabs.com/platform/platform-configurator#/) tool for the most up-to-date information.
```

| iPad (10th Generation) Simulator + `latest` or `current_major` |            16.1-16.4             |        16.1-16.4        |   16.4   |                                                                                                                                                                                                        |
| iPad (10th Generation) Simulator + `previous_major`            |            16.1-16.4             |        16.1-16.4        | &#x2715; | <pre>Error: Failed to create session. <br/>Invalid platform version specified for simulator:<br/>iPad (10th Generation) Simulator</pre> iPad (10th Generation) was released with iOS 16.1, not with 15 |
| iPhone 13 Simulator + `latest` or `current_major`              |            15.0-16.4             |        15.0-16.4        |   16.4   | iPhone 13 was released with iOS 15 and supports 16                                                                                                                                                     |

> | iPhone 13 Simulator + `latest` or `current_major`              |            15.0-16.4             |        15.0-16.4        |   16.4   | iPhone 13 was released with iOS 15 and supports 16                                                                                                                                                     |

Author: Thanks!
```suggestion
```

> | iPhone 13 Simulator + `latest` or `current_major`              |            15.0-16.4             |        15.0-16.4        |   16.4   | iPhone 13 was released with iOS 15 and supports 16                                                                                                                                                     |

Reviewer: Repetition of row.
